src/RequestHandler/outofprocess/serverprocess.cxx

```diff
@@ -887,6 +905,12 @@ SERVER_PROCESS::StartProcess(
         goto Finished;
 
     Failure:
+        if (fCriticalError)
+        {
+            // Critical error, no retry need to avoid wasting resource and polluting log
+            dwRetryCount = 0;
```

Comment:
Is there any other failure cases we would consider "critical" here? If we are going to pass in a bool for setting up the port, should we pass it in to other functions?